v2/internal/controllers/to_azure_secrets_test.go

```diff
@@ -79,10 +80,118 @@ func Test_MissingSecretKey_ReturnsError(t *testing.T) {
 	// We expect the ready condition to include details of the error
-	tc.Expect(vm.Status.Conditions[0].Reason).To(Equal("SecretNotFound"))
+	tc.Expect(vm.Status.Conditions[0].Reason).To(Equal(conditions.ReasonSecretNotFound))
 	tc.Expect(vm.Status.Conditions[0].Message).To(
 		ContainSubstring("Secret \"%s/%s\" does not contain key \"%s\"", tc.Namespace, secret.Name, secret.Key))
 
 	// Delete VM and resources.
 	tc.DeleteResourceAndWait(rg)
 }
+
+func Test_UserSecretInDifferentNamespace_SecretNotFound(t *testing.T) {
+	t.Parallel()
+
+	tc := globalTestContext.ForTest(t)
+	secretName := tc.Namer.GenerateName("admin-pw")
+	nameSpaceName := "secret-namespace"
+
+	createNamespaces(tc, nameSpaceName)
+	secretInDiffNamespace := createNamespacedSecret(tc, nameSpaceName, secretName)
+
+	rg := tc.CreateTestResourceGroupAndWait()
+
+	vnet := newVMVirtualNetwork(tc, testcommon.AsOwner(rg))
+	subnet := newVMSubnet(tc, testcommon.AsOwner(vnet))
+	networkInterface := newVMNetworkInterface(tc, testcommon.AsOwner(rg), subnet)
+
+	tc.CreateResourceAndWait(vnet)
+	tc.CreateResourceAndWait(subnet)
+	tc.CreateResourceAndWait(networkInterface)
+
+	vm := newVirtualMachine20201201(tc, rg, networkInterface, secretInDiffNamespace)
+
+	tc.CreateResourceAndWaitForState(vm, metav1.ConditionFalse, conditions.ConditionSeverityWarning)
+	// We expect the ready condition to include details of the error
+	tc.Expect(vm.Status.Conditions[0].Reason).To(Equal(conditions.ReasonSecretNotFound))
+	tc.Expect(vm.Status.Conditions[0].Message).To(
+		ContainSubstring("failed resolving secret references: %s/%s does not exist", tc.Namespace, secretInDiffNamespace.Name))
+
+	// Delete VM and resources.
+	tc.DeleteResourcesAndWait(vm, networkInterface, subnet, vnet, rg)
+}
+
+func Test_UserSecretInDifferentNamespace_ShouldNotTriggerReconcile(t *testing.T) {
+	t.Parallel()
+	ns1 := "secret-in-different-namespace-ns-1"
+	ns2 := "secret-in-different-namespace-ns-2"
+
+	cfg, err := testcommon.ReadFromEnvironmentForTest()
+	if err != nil {
+		t.Fatal(err)
+	}
+	cfg.TargetNamespaces = []string{ns1, ns2}
+	tc := globalTestContext.ForTestWithConfig(t, cfg)
+
+	createNamespaces(tc, ns1, ns2)
+
+	secretName := tc.Namer.GenerateName("server-secret")
+	ns1Secret := createNamespacedSecret(tc, ns1, secretName)
+	ns2Secret := createNamespacedSecret(tc, ns2, secretName)
+
+	rg := tc.NewTestResourceGroupWithNamespace(ns1)
+	tc.CreateResourceGroupAndWait(rg)
+
+	rg2 := tc.NewTestResourceGroupWithNamespace(ns2)
+	tc.CreateResourceGroupAndWait(rg2)
+
+	server1, _ := newFlexibleServer(tc, rg, ns1Secret)
+	server1.ObjectMeta = metav1.ObjectMeta{
+		Name:      tc.Namer.GenerateName("mysql"),
+		Namespace: ns1,
+	}
+	tc.CreateResourceAndWait(server1)
+
+	resourceID := genruntime.GetResourceIDOrDefault(server1)
+	tc.Expect(resourceID).ToNot(BeEmpty())
+
+	// Deleting server1 here using AzureClient so that Operator does not know about the deletion
```

Comment:
Minor: suggest expanding this to: "Deleting server1 here using AzureClient so that Operator does not know about the deletion. If operator reconciles that resource again it will be recreated and we will notice"

Reply:
Done!